warehouse/admin/views/users.py

```diff
@@ -92,6 +93,14 @@ class UserForm(forms.Form):
 
     emails = wtforms.fields.FieldList(wtforms.fields.FormField(EmailForm))
 
+    def validate_emails(self, field):
+        # If there's no email on the account, it's ok. Otherwise, ensure
+        # we have 1 primary email.
+        if field.data and len([1 for email in field.data if email["primary"]]) != 1:
```

Comment:
Here's a fun trick
```suggestion
        primaries = iter([email["primary"] for email in field.data])
        if any(primaries) and not any(primaries):
```
I'm not sure that's more readable than what you had here, but it came to mind when reviewing.

Reply:
Haha, fun. But why do `iter([a for a in b])` when you can do `(a for a in b)` ? 😄 

It's fun, it's also less readable IMHO. (also, if there's no email at all we should be forced to add one, which I believe your example doesn't do) ((also I think you meant it the other way around, we want to `not` this whole clause so that's `if not any(primaries) or any(primaries)`)

Other solutions (that also don't handle the no email case for the sake of simplicity):
```python
import collections
if collections.Counter(email["primary"] for email in field.data).get(True) != 1:
```
Or good ol':
```
# bools are just integers with values 0 or 1
if sum(email["primary"] for email in field.data) != 1:
```
I actually quite like the last one :o

Reply:
```suggestion
        if field.data and sum(email["primary"] for email in field.data) != 1:
```